src/textual/widgets/_month_calendar.py

```diff
@@ -0,0 +1,107 @@
+from __future__ import annotations
+
+import calendar
+import datetime
+
+from rich.text import Text
+from textual.app import ComposeResult
+from textual.reactive import Reactive, var
+from textual.widget import Widget
+from textual.widgets import DataTable
+
+
+class InvalidMonthNumber(Exception):
+    pass
+
+
+class InvalidWeekdayNumber(Exception):
+    pass
+
+
+class MonthCalendar(Widget):
+    year: Reactive[int | None] = Reactive[int | None](None)
+    month: Reactive[int | None] = Reactive[int | None](None)
+    first_weekday: Reactive[int] = Reactive(0)
```

Comment:
I'm wondering about using `Literal["monday", "tuesday", ...]` here to remove any second-guessing about how days are indexed. Will leave it as an open thought here if you or anyone else wants to chip in.

Reply:
Perhaps this could be an `IntEnum` [similar to the calendar module?](https://github.com/Textualize/textual/pull/3667#discussion_r1392221606)